docs/configuration/https.md

```diff
@@ -0,0 +1,65 @@
+# HTTPS and authentication
+
+Prometheus supports basic authentication and TLS.
+This is **experimental** and might change in the future.
+
+To enable TLS or basic authentication, use the flag `--web.config.file`.
+
+e.g. `./prometheus --web.config.file="web-config.yml"`
+
+The configuration file should be written in YAML format, and is reloaded on each
+connection to check for new certificates and/or authentication policy.
+
+Generic placeholders definition can be found in the [configuration
```

Comment:
Let's keep this self-contained, it's what we do elsewhere plus this there's a good change this will move out of Prometheus-specific docs.

Reply:
Good idea. I have also taken the same phrasing.